### PR TITLE
mailmanPackages: update

### PR DESCRIPTION
**Pull request by Ma27**
## Description of changes

Updates the components with newer releases available. Also cleaned up the legacy `psycopg2` hack.

Running productively on my employer's mailman instance.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/mail/mailman/python.nix [hunk 1]
```diff
@@ -1,19 +1,12 @@
...
-        inherit version;
-        inherit (a) pname;
-        sha256 = "fb23f6c71107c37fd667cb4ea363ddeb936b348bbd6449278eb92c189699f543";
-      };
-    });
-  };
+  packageOverrides = lib.composeExtensions
+    (self: super: {
+      # does not find tests
+      alembic = super.alembic.overridePythonAttrs (oldAttrs: {
```
Comment (kirillrdy) [inline pkgs/servers/mail/mailman/python.nix:7]: this is no longer necessary

File: pkgs/servers/mail/mailman/python.nix [hunk 2]
```diff
@@ -1,19 +1,12 @@
-{ python3, fetchPypi }:
+{ python3, lib, overlay ? (_: _: {}) }:
```
Comment (kirillrdy) [inline pkgs/servers/mail/mailman/python.nix:1]: optional: python.nix is not longer necessary

**Review by kirillrdy — APPROVED** [review]

**Comment by kirillrdy** [comment]
```
The underlying issue is that the `packageOverrides`-mechanism of
`pkgs.python3` doesn't compose, so an optional overlay is manually
applied to the `python3` used for mailman.
```

I am not sure if its related, but when using `packageOverrides` you also need to override `self` for python

eg
```
localPython = python3.override {
      self = localPython;
      packageOverrides = self: super: {
     }
}
```

**Comment by Ma27** [comment]
@kirillrdy I actually think it's better to keep the `python.nix` because I'd expect that we'll need this kind of setup in the future again and I don't really want to change this back and forth whenever we need to override dependencies. However, when not using it, people will probably make the mistake of forgetting `self =` again and then will be confused why their overrides suddenly stop working. IMHO the correct fix would be to implement overlays on the python package-set, then we can remove the `overlay` argument.

Until then, this is IMHO a good way to provide a stable API to always override dependencies of mailman.

Good catch with alembic btw, I removed that one entirely.

**Comment by kirillrdy** [comment]
yes I understand and agree :-)

**Review by kirillrdy — APPROVED** [review]

**Comment by Ma27** [comment]
Splendid, thanks for looking over this!

**Pull request merged**